api/v1beta1/modelregistry_types.go

```diff
@@ -56,22 +56,23 @@ type PostgresConfig struct {
 	// Port number to connect to at the server host.
 	Port *int32 `json:"port,omitempty"`
 
-	//+kubebuilder:required
 	// PostgreSQL username to connect as.
 	Username string `json:"username,omitempty"`
 
 	// Password to be used if required by the PostgreSQL server.
 	PasswordSecret *SecretKeyValue `json:"passwordSecret,omitempty"`
 
-	//+kubebuilder:required
 	// The database name.
-	Database string `json:"database"`
+	Database string `json:"database,omitempty"`
 
 	//+kubebuilder:default=false
 	// True if skipping database instance creation during ML Metadata
 	// service initialization. By default, it is false.
 	SkipDBCreation bool `json:"skipDBCreation,omitempty"`
 
+	// Auto-provision a PostgreSQL database if true.
+	Generate *bool `json:"generate,omitempty"`
```

Comment:
Can this property be something more specific? It seems ambiguous about what's being generated.
Maybe something like `defaultDeployment`, or `generateDeployment` so it's explicit.